PGM/src/main/java/tc/oc/pgm/mutation/types/kit/TeamChestMutation.java

```diff
@@ -0,0 +1,139 @@
+        Optional<MatchPlayer> optPlayer = match().participant((Entity) bukkitPlayer);
+        if (!optPlayer.isPresent() ||
+                !(event.getAction() == Action.RIGHT_CLICK_AIR || event.getAction() == Action.RIGHT_CLICK_BLOCK) ||
+                event.getItem() == null ||
+                event.getItem().getType() != TOOL_TYPE) {
+            return;
+        }
+
+        Optional<Inventory> optTeamInventory = getTeamsInventory(bukkitPlayer);
+        optTeamInventory.ifPresent(teamInventory -> {
+            event.setCancelled(true);
+            // If the item is in the off-hand slot, it wont get put back visually for the player without this.
+            if(event.getHand() == EquipmentSlot.OFF_HAND) event.getActor().updateInventory();
+            bukkitPlayer.openInventory(teamInventory);
+        });
+    }
+
+    @EventHandler(priority = EventPriority.LOWEST)
+    public void onInventoryClick(InventoryClickEvent event) {
+        Player player = event.getActor();
+        if (event.getCurrentItem() == null) return;
+
+        // No putting blacklisted items (ender chest, possibly wool) into the chest
+        Optional<Inventory> teamChest = getTeamsInventory(player);
+        if (teamChest.map(teamInventory -> teamInventory.equals(event.getView().getTopInventory())).orElse(false) &&
+                isBlacklistedItem(event.getCurrentItem())) {
+            event.setCancelled(true);
+            return;
+        }
+
+        // If normal right click, in their inventory, on the chest, then open shared inventory.
+        getTeamsInventory(player).ifPresent(teamInventory -> {
+            if (event.getInventory().getType() == InventoryType.CRAFTING &&
+                    event.getCurrentItem().getType() == TOOL_TYPE &&
+                    event.getAction() == InventoryAction.PICKUP_HALF) {
+                event.setCancelled(true);
+                // This resets their mouse position annoyingly, but without it items can get stuck in places.
+                player.closeInventory();
+                // Prevent visual inconsistencies, specifically off-hand slot
+                if (event.getSlotType() == InventoryType.SlotType.QUICKBAR) {
+                    player.updateInventory();
+                }
+                player.openInventory(teamInventory);
+            }
+        });
+    }
+
+    private boolean isBlacklistedItem(ItemStack item) {
+        return item.getType() == TOOL_TYPE ||
+                optWools.map(w -> w.isObjectiveWool(item)).orElse(false);
+    }
+
+    private Optional<Inventory> getTeamsInventory(Player bukkitPlayer) {
+        return match().participant((Entity) bukkitPlayer)
+                .map(matchPlayer -> teamChests.get(matchPlayer.getParty()));
+    }
+
+    private Kit getKitForPlayer(MatchPlayer player) {
+        ItemStack stack = new ItemBuilder(item(TOOL_TYPE))
+                .name(ChatColor.DARK_PURPLE + PGMTranslations.t("mutation.type.teamchest.item_name", player))
```

Comment:
Are we able to not use PGMTranslations?

Reply:
I couldn't seem to get it to work yet,
`new Component(new TranslatableComponent("mutation.type.teamchest.item_name"),
                        ChatColor.DARK_PURPLE).getText()`
doesn't work.

Reply:
toLegacyText()

Reply:
How do I get a `ComponentRenderContext`? That is what I assume I need (Readme says so), because nothing is translating it is just showing the translation key.

Reply:
Mutations don't support injection right now, so using `PGMTranslations` should be fine.